dashboard/test/controllers/data_docs_controller_test.rb

```diff
@@ -84,4 +90,28 @@ class DataDocsControllerTest < ActionController::TestCase
     get :edit, params: {key: 'unknown_key'}
     assert_response :not_found
   end
+
+  test 'data is passed to edit_all page' do
+    sign_in @levelbuilder
+
+    get :edit_all
+    assert_response :ok
+
+    show_data = css_select('script[data-datadocs]').first.attribute('data-datadocs').to_s
+    assert_equal DataDoc.all.order(:name).map(&:serialize).to_json, show_data
+  end
+
+  test 'data doc is deleted through destroy route' do
+    sign_in @levelbuilder
+    data_doc_to_delete = create :data_doc, key: 'doc_to_delete'
+
+    post :destroy, params: {
+      key: data_doc_to_delete.key
+    }
+    assert_response :no_content
```

Comment:
This is making me wonder if we want to try something like
https://github.com/code-dot-org/code-dot-org/blob/d3d1123b7851d12704f89658c4d3e8632f403958/dashboard/app/controllers/programming_expressions_controller.rb#L113-L118

partly so that we can `assert_response :ok` here

Reply:
Also partly so that we can show an error message on the frontend if something didn't work right

Reply:
will do!